stackrox/stackrox
# ROX-12834: Introduce fake nodescan generator

## Description

This PR introduces a generator to send fake NodeScanV2 messages from Collector to Sensor.
These messages enable us to build out the full pipeline independently from implementing the NodeScanV2 logic.

* This feature is hidden behind a feature flag. Use `ROX_RHCOS_NODE_SCANNING=true` when calling `deploy.sh` to enable it.
* The interval for scans is already customizable through an env var, `ROX_NODE_SCAN_INTERVAL` that defaults to 4 hours
  * The interval can be decreased by setting this var to another value for the Collector image in the Collector DaemonSet 


## Checklist

If any of these don't apply, please comment below.

## Testing Performed

1. Deploy (e.g. on local) with enabled feature flag
`ROX_RHCOS_NODE_SCANNING=true ./deploy/k8s/deploy-local.sh`
2. Observe the log message in the compliance container (in the compliance pod) that the scanning interval is 4 hours
3. Observe a fake Node Scan message being logged in Sensor logs once at start, then every 4 hours
~~4. Edit Collector DaemonSet and add a custom interval as env var to the collector image~~

4. Set environment variable via `kubectl`: `kubectl -n stackrox set env daemonsets/collector --containers="compliance" ROX_NODE_RESCAN_INTERVAL="20s"` 

5. Observe the Collector Pod getting reinstantiated and the collector image log message stating an interval of 15 seconds.

6. Finally, `teardown` everything and redeploy with `deploy.sh` without setting the feature flag.
Observe the collector log not stating a Node Scan Interval.


### Additional testing: Resilience
We also performed error cases:
1. Kill Sensor pod, observe Compliance container re-establishing communication and continue sending messages
2. Kill Compliance pod, observe Compliance container re-establish communication and continue sending messages
